addons/membership/membership.py

```diff
@@ -149,7 +151,7 @@ def _state(self, cr, uid, ids, name, args, context=None):
         'member_price': fields.float('Membership Fee', digits_compute= dp.get_precision('Product Price'), required=True, help='Amount for the membership'),
         'account_invoice_line': fields.many2one('account.invoice.line', 'Account Invoice line', readonly=True),
         'account_invoice_id': fields.related('account_invoice_line', 'invoice_id', type='many2one', relation='account.invoice', string='Invoice', readonly=True),
-        'state': fields.function(_state,
+        'state': fields.function(__get_state,
```

Comment:
This is a good point, thanks for your answer.
I've test it, and works perfectly. This could be apply in other similar cases.